### PR TITLE
Add clippy to all rust projects

### PR DESCRIPTION
**Pull request by ali-bahjati**
Also enable auto fix option as it solves most of the obvious problems.

I advice at least one person review the mostly automated changes to the files.

Please note that I made some changes manually, for example `pyth_client` was deprecated and I changed it to use `pyth_sdk_solana::state`.

### CODE REVIEW

File: solana/pyth2wormhole/client/Cargo.toml [hunk 1]
```diff
@@ -39,6 +39,5 @@ warp = "0.3.3"
 http = "0.2.8"
 
 [dev-dependencies]
-pyth-client = "0.5.0"
```
Comment (drozdziak1) [inline solana/pyth2wormhole/client/Cargo.toml:42]: This is wrong! It is used in the standalone tests. I think there should be a way to tell clippy to take `--test` into account.

Comment (drozdziak1) [reply solana/pyth2wormhole/client/Cargo.toml:42]: I think our Cargo caching may be giving us a false positive about this.

Comment (drozdziak1) [reply solana/pyth2wormhole/client/Cargo.toml:42]: example: test_attest.rs

File: solana/pyth2wormhole/client/src/util.rs [hunk 2]
```diff
@@ -84,7 +84,7 @@ impl<T> RLMutex<T> {
     pub fn new(val: T, rl_interval: Duration) -> Self {
         Self {
             mtx: Mutex::new(RLMutexState {
-                last_released: Instant::now() - rl_interval,
+                last_released: Instant::now().checked_sub(rl_interval).unwrap(),
```
Comment (drozdziak1) [inline solana/pyth2wormhole/client/src/util.rs:87]: I guess? Underflows should be impossible for any reasonable value (i.e. lower than current unix timestamp value, which is decades worth)

File: solana/pyth2wormhole/client/tests/fixtures/pyth.rs [hunk 3]
```diff
@@ -1,11 +1,10 @@
 //! This module contains test fixtures for instantiating plausible
 //! Pyth accounts for testing purposes.
 use {
-    pyth_client::{
-        AccKey,
+    pyth_sdk_solana::state::{
         AccountType,
-        Price,
-        Product,
+        PriceAccount,
```
Comment (drozdziak1) [inline solana/pyth2wormhole/client/tests/fixtures/pyth.rs:6]: I'm not sure that I understand clippy's reasoning here. Hos does it make this decision?

File: solana/pyth2wormhole/program/src/config.rs [hunk 4]
```diff
@@ -154,7 +154,7 @@ impl From<Pyth2WormholeConfigV2> for Pyth2WormholeConfigV3 {
             wh_prog,
             pyth_owner,
             max_batch_size,
-            is_active,
+            is_active: _,
```
Comment (drozdziak1) [inline solana/pyth2wormhole/program/src/config.rs:157]: I think it's wrong that we forced is_active to `true` with this, but admittedly this is being used rarely enough that it shouldn't matter

File: solana/pyth2wormhole/program/src/lib.rs [hunk 5]
```diff
@@ -1,3 +1,4 @@
+#![allow(incomplete_features)]
```
Comment (drozdziak1) [inline solana/pyth2wormhole/program/src/lib.rs:1]: I think this is necessary for  the `adt_const_params` just below it in newer nightlies. Nice!

**Review by drozdziak1 — CHANGES_REQUESTED** [review]
Found some nits but I think the useful things are in overwhelming majority.

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -17,29 +17,56 @@ repos:
...
         entry: cargo +nightly fmt --manifest-path ./pythnet/remote-executor/Cargo.toml --all -- --config-path rustfmt.toml
         pass_filenames: false
         files: pythnet/remote-executor/
-      - id: cargo-clippy-executor
-        name: Cargo clippy executor
+      - id: cargo-clippy-remote-executor
+        name: Cargo clippy for remote executor
         language: "rust"
-        entry: cargo +nightly clippy --manifest-path ./pythnet/remote-executor/Cargo.toml -- -D warnings
+        entry: cargo +nightly clippy --manifest-path ./pythnet/remote-executor/Cargo.toml --fix --allow-dirty -- -D warnings
```
Comment (guibescos) [inline .pre-commit-config.yaml:29]: Does this run in the tests as well?

File: solana/pyth2wormhole/client/Cargo.toml (hunk 1 above)
Comment (ali-bahjati) [reply solana/pyth2wormhole/client/Cargo.toml:42]: That one I removed manually, sorry that I forgot to mention it myself.
Clippy was complaining that the package is deprecated (we marked all its method deprecated). I replaced it with pyth_sdk_solana

File: solana/pyth2wormhole/client/src/util.rs (hunk 2 above)
Comment (ali-bahjati) [reply solana/pyth2wormhole/client/src/util.rs:87]: yeah I checked at that is nearly impossible to fail. So I think unwrap is fine.

File: solana/pyth2wormhole/client/tests/fixtures/pyth.rs (hunk 3 above)
Comment (drozdziak1) [reply solana/pyth2wormhole/client/tests/fixtures/pyth.rs:6]: Ohhh, I get it now, these are the same structs but in the changed package. I think that's alright!

**Review by drozdziak1 — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 6 above)
Comment (ali-bahjati) [reply .pre-commit-config.yaml:29]: added it.

File: solana/pyth2wormhole/program/src/config.rs (hunk 4 above)
Comment (ali-bahjati) [reply solana/pyth2wormhole/program/src/config.rs:157]: yeah. it does not matter now anymore.

**Review by jayantk — COMMENTED** [review]
seems like you got some reviews here so i'm not going to add one of my own

**Pull request merged**